src/__swaps__/screens/Swap/components/SearchInput.tsx

```diff
@@ -55,25 +58,26 @@ export const SearchInput = ({
   const labelQuaternary = useForegroundColor('labelQuaternary');
 
   const btnText = useDerivedValue(() => {
-    if ((inputProgress.value === 2 && !output) || (outputProgress.value === 2 && output)) {
+    if (
+      (inputProgress.value === NavigationSteps.SEARCH_FOCUSED && !output) ||
+      (outputProgress.value === NavigationSteps.SEARCH_FOCUSED && output)
+    ) {
       return CANCEL_LABEL;
     }
 
     if ((output && internalSelectedOutputAsset.value) || !output) {
       return CLOSE_LABEL;
     }
 
-    // ⚠️ TODO: Add paste functionality to the asset to buy list when no asset is selected
-    // return PASTE_LABEL;
+    return PASTE_LABEL;
   });
 
   const buttonVisibilityStyle = useAnimatedStyle(() => {
-    const isSearchFocused = (output ? outputProgress : inputProgress).value === NavigationSteps.SEARCH_FOCUSED;
-    const isAssetSelected = output ? internalSelectedOutputAsset.value : internalSelectedInputAsset.value;
+    const isVisible = output || inputProgress.value === NavigationSteps.SEARCH_FOCUSED || internalSelectedOutputAsset.value;
```

Comment:
I wouldn't get rid of the other conditionals here, instead I would just add another which is basically the same check you have in the `isPaste` below:

```
output && outputProgress.value === NavigationSteps.TOKEN_LIST_FOCUSED && !internalSelectedOutputAsset.value
```

Reply:
if its output it's always visible, I can do it like this
```ts
const isInputSearchFocused = inputProgress.value === NavigationSteps.SEARCH_FOCUSED;
const isInputAssetSelected = !!internalSelectedOutputAsset.value
const isVisible = output || isInputSearchFocused || isInputAssetSelected;
```
just removing the output cases since it wont matter just makes it more confusing